WordPress/Classes/Networking/ServiceRemoteREST.m

```diff
@@ -0,0 +1,20 @@
+#import "ServiceRemoteREST.h"
+#import "WordPressComApi.h"
+
+@interface ServiceRemoteREST ()
+@end
+
+@implementation ServiceRemoteREST
+
+- (id)initWithApi:(WordPressComApi *)api {
+    
+    NSAssert([api isKindOfClass:[WordPressComApi class]],
```

Comment:
I'm a little bit more inclined to `NSAssert`, since we're checking for the api's kind. `NSParameterAssert` with `isKindOfClass` would feel odd. And ditching `isKindOfClass` would be less effective (imho!)